benchmark/operations/create_container.py

```diff
@@ -39,11 +43,14 @@ def find_match(states):
 
         return d
 
-    return loop_until(reactor, state_reached)
+    d = loop_until(reactor, state_reached)
+    _timeout(reactor, d, timeout)
```

Comment:
:question: Another way of implementing a timeout is to pass a finite iterator to `loop_until`. e.g. 

```python
d = loop_until(reactor, state_reached, [0.1] * DEFAULT_TIMEOUT / 0.1)
```

Obviously this isn't _quite_ the same thing, since it doesn't account for the probe time.